src/GuzzleHttp/Middleware/HistoryMiddleware.php

```diff
@@ -23,34 +23,32 @@ class HistoryMiddleware
 {
     private $container;
 
-    public function __construct(\SplObjectStorage $container)
+    public function __construct(\ArrayObject $container)
     {
         $this->container = $container;
     }
 
     public function __invoke(callable $handler)
     {
         return function (RequestInterface $request, array $options) use ($handler) {
+            $correlationId = uniqid();
+            $request = $request->withAddedHeader('csa-guzzle-correlation-id', $correlationId);
```

Comment:
Move the header name in a class constant, and add the `X-` prefix. This is a non-standard header, and should be prefixed by `X-` (eXtended).

See https://github.com/csarrazi/CsaGuzzleBundle/blob/548e35124e4e854f610bcc501d239c6cbda23316/src/GuzzleHttp/Middleware/CacheMiddleware.php#L26 for more info.